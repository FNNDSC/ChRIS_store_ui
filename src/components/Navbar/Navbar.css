header {
  /* shadow */
  box-shadow: 0 20px 20px -20px rgba(0, 0, 0, 0.5);
}

nav.navbar-default {
  /* display */
  height: 132px;

  /* position */
  position: relative;
}

div.navbar-row {
  /* display */
  display: flex;
}

div.navbar-row > .navbar-header {
  /* flex */
  flex: 1;
}

header .row > .navbar-header > .navbar-logo {
  /* display */
  padding: 0 !important;
  margin: 8px 0;

  /* other */
  outline: 0;
}

.navbar-logo-img {
  /* display */
  width: 243px;
  height: 48px;
}

.navbar-header > .navbar-search.search-pf {
  /* color */
  color: var(--pf-black-200);
  
  /* display */
  display: block;
  
  /* position */
  position: absolute;
  top: 66px;
  left: 0;
  right: 0;
}

.navbar-header > .navbar-search.search-pf > .form-group {
  /* display */
  display: block;
}

.navbar-search > .has-icon .search-icon {
  /* display */
  display: flex;
  width: 64px;
  height: 64px;

  /* font */
  font-size: 1.5em;

  /* flex */
  flex-direction: column;
  justify-content: center;
}

.navbar-search .form-group .form-control {
  /* color */
  color: var(--pf-black-200);
  background: var(--pf-black-900);

  /* display */
  padding: 32px 66px;

  /* font */
  font-size: 1.2em;
  font-weight: 600;

  /* border */
  border: 0;
  border-radius: 0;

  /* other */
  box-shadow: none;
}

.navbar-search .form-control::placeholder {
  /* color */
  color: var(--pf-black-200);

  /* font */
  font-style: normal;
}

.navbar-trigger {
  /* color */
  color: var(--pf-black-200);

  /* display */
  margin: 12px 15px;
  padding: 0 10px;

  /* font */
  font-size: 2em;
  
  /* other */
  float: right;
  cursor: pointer;
}

/* ===== INACTIVE NAV LINKS ===== */

header .navbar.navbar-default .navbar-right > li > a:hover {
  /* color */
  color: var(--pf-black-300);
}

header .navbar.navbar-default .navbar-right > li > a {
  /* color */
  color: var(--pf-black-100);

  /* display */
  padding: 22px 15px;

  /* font */
  font-size: 1.2em;
}

header .navbar.navbar-default .navbar-right > li > button {
  /* display */
  margin: 16px 13px;
}

/* ====== ACTIVE NAV LINKS ====== */

header .navbar.navbar-default .navbar-right > li > a.active {
  /* color */
  color: var(--pf-black-100);

  /* other */
  cursor: default;
}

header .navbar.navbar-default .navbar-right > li > a.active:hover {
  /* color */
  background: inherit;
}

/* ======= NAVBAR DROPDOWN ====== */

header > .navbar-dropdown {
  /* color */
  background: #1d1d1d;

  /* border */
  border-bottom: 1px solid var(--pf-black-800);

  /* other */
  z-index: 1;
}

.navbar-btn-container {
  /* font */
  line-height: 1;
}

/* ==== INACTIVE DROPDOWN BTN === */

.navbar-dropdown-btn {
  /* display */
  display: block;
  padding: 0.5em;
  
  /* color */
  color: var(--pf-black-100);

  /* font */
  font-size: 2em;
  font-weight: 600;
}

.navbar-dropdown-btn:hover {
  /* color */
  background: var(--pf-black-900);
}

/* ===== ACTIVE DROPDOWN BTN ==== */

.navbar-dropdown-btn.active {
  /* color */
  background: var(--pf-black);
  color: var(--pf-black-100);

  /* other */
  cursor: default;
}

.navbar-btn-container .navbar-dropdown-btn:hover,
.navbar-btn-container .navbar-dropdown-btn:focus {
  color: var(--pf-black-100);
  text-decoration: none;
}

/* ========= RESPONSIVE ========= */

@media (min-width: 480px) {
  header .navbar-header > .navbar-logo {
    /* display */
    margin: 0 0 0 25px;
  }
}

@media (min-width: 768px) {
  nav.navbar-default { height: 66px; }

  .navbar-header > .navbar-search.search-pf { 
    /* color */
    color: inherit;

    /* display */
    display: table;
    margin: 19px 0;
    padding-left: 1em;
    
    /* position */
    position: static; 
  }

  .navbar-search .form-group {
    /* position */
    position: relative;

    /* display */
    margin: 0;
  }

  .navbar-search > .has-icon .search-icon {
    /* color */
    color: var(--pf-black-300);

    /* display */
    width: 1.35em;
    height: 1.35em;
  
    /* font */
    font-size: 1.5em;
  }

  .navbar-search .form-group .form-control {
    /* color */
    background: inherit;
    color: var(--pf-black-300);

    /* display */
    padding: 0 0 0 2em;

    /* font */
    font-size: 1.3em;
    font-weight: 600;
  }

  .navbar-search .form-control::placeholder {
    /* color */
    color: var(--pf-black-300);
  }

  .navbar-trigger, .navbar-dropdown { display: none; }
  .collapse.in { display: none !important; }
}

@media (min-width: 1110px) {
  header .row > .navbar-header > .navbar-logo {
    /* display */
    margin-left: 0;
  }
}
