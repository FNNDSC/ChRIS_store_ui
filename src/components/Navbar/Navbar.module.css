.login-button .other-button{
  background: #06c !important;
  color: white !important;
}