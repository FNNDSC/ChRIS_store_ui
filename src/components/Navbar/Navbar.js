import React, { useState } from 'react';
import {
  Brand,
  PageHeader,
  Nav,
  NavList,
  NavItem,
  PageHeaderTools,
  PageHeaderToolsItem,
} from '@patternfly/react-core';
import Button from '../Button';
import { NavLink, useHistory } from 'react-router-dom';
import Search from './components/Search/Search';
import './Navbar.css';
import LogoImg from '../../assets/img/chris-plugin-store_logo.png';
import ChrisStore from '../../store/ChrisStore';

const navLinks = [
  {
    label: 'Plugins',
    to: '/plugins'
  },
  {
    label: 'Quick Start',
    to: '/quickstart'
  },
  {
    label: 'Dashboard',
    to: '/dashboard',
    cond: (store) => store.get('isLoggedIn')
  }
];

/**
 * Conditionally renders a list of links into a <Nav>.
 */
const Navigation = ({ store }) => {
  const [searchKey, setSearchKey] = useState('');
  const history = useHistory();
  const shouldShowLink = (linkInfo) => {
    if (!linkInfo.cond) {
      return true;
    }
    return linkInfo.cond(store);
  };
  return (
    <Nav variant="horizontal">
      <NavList>
        {
          navLinks
            .filter((l) => shouldShowLink(l))
            .map(link => (
              <NavItem
                key={link.to}
                itemId={link.to}
                isActive={window && window.location.pathname === link.to}>
                <NavLink to={link.to} activeClassName="pf-m-current">
                  {link.label}
                </NavLink>
              </NavItem>
            ))
        }
        <NavItem>
          <Search 
            placeholder="Search Plugin"
            value={searchKey}
            onChange={(value) => setSearchKey(value)}
            onClear={() => setSearchKey('')}
            onSearch={() => {history.push('/plugins')}}
          />
        </NavItem>
      </NavList>
    </Nav>
  );
}

const StatefulNavigation = ChrisStore.withStore(Navigation);
const statefulNavigation = (<StatefulNavigation />);

const LoginButton = ({ store }) => (
  <NavLink to="/signin">
    <Button
      variant="primary"
      className="login-button"
    >
      {store.get('isLoggedIn') ? 'Sign Out' : 'Sign In'}
    </Button>
  </NavLink>
);
const StatefulLoginButton = ChrisStore.withStore(LoginButton);

const HeaderTools = (
  <PageHeaderTools>
    <PageHeaderToolsItem>
      <StatefulLoginButton />
    </PageHeaderToolsItem>
  </PageHeaderTools>
)

const Logo = (<Brand alt="ChRIS Plugin Store" src={LogoImg}/>);

const Navbar = () => (
  <PageHeader
    logo={Logo}
    logoComponent={NavLink}
    logoProps={{to: '/'}}
    topNav={statefulNavigation}
    headerTools={HeaderTools}
  />
);

export default Navbar;
