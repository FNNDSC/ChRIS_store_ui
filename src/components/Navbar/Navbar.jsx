import React from 'react';
import {
  Button,
  Brand,
  PageHeader,
  Nav,
  NavList,
  NavItem,
  PageHeaderTools,
  PageHeaderToolsItem,
} from '@patternfly/react-core';
import { NavLink } from 'react-router-dom';
import './Navbar.css';
import LogoImg from '../../assets/img/chris-plugin-store_logo.png';
import ChrisStore from '../../store/ChrisStore';

const navLinks = [
  {
    label: 'Plugins',
    to: '/plugins',
  },
  {
    label: 'Pipelines',
    to: '/pipelines',
  },
  {
    label: 'Submit your Plugin',
    to: '/quickstart',
  },
  
  {
    label: 'Dashboard',
    to: '/dashboard',
    if: (store) => store.get('isLoggedIn')
  }
];

/**
 * Conditionally renders a list of links into a <Nav>.
 */
const Navigation = ChrisStore.withStore(({ store }) => (
  <Nav variant="horizontal">
    <NavList>
      {
        navLinks
          .filter(link => {
            if (!link.if)
              return true;
            return link.if(store);
          })
          .map(link => (
            <NavItem
              key={link.to}
              itemId={link.to}
              isActive={window && window.location.pathname === link.to}>
              <NavLink to={link.to} activeClassName="pf-m-current">
                {link.label}
              </NavLink>
            </NavItem>
          ))
      }
    </NavList>
  </Nav>
));

const Navbar = ({ searchComponent, store }) => {
  const HeaderTools = (
    <PageHeaderTools>
      <PageHeaderToolsItem>
        {searchComponent}
      </PageHeaderToolsItem>
      <PageHeaderToolsItem>
        <NavLink to="/signin">
          <Button id="login-button">
            {store.get('isLoggedIn') ? 'Sign Out' : 'Sign In'}
          </Button>
        </NavLink>
      </PageHeaderToolsItem>
    </PageHeaderTools>
  )

  const Logo = (<Brand className="logo" alt="ChRIS Plugin Store" src={LogoImg}/>)
  
  return (
    <PageHeader
      logo={Logo}
      logoComponent={NavLink}
      logoProps={{to: '/'}}
      topNav={(<Navigation />)}
      headerTools={HeaderTools}
    />
  )
}
export default ChrisStore.withStore(Navbar);
