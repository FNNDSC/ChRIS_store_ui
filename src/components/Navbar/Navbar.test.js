import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar';

describe('Navbar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Navbar />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render nav element', () => {
    expect(wrapper.find('nav')).toHaveLength(1);
  });

  it('should render navbar-row div inside nav', () => {
    expect(wrapper.find('div.navbar-row')).toHaveLength(1);
  });

  it('should render 5 NavLink components', () => {
    expect(wrapper.find('NavLink')).toHaveLength(5);
  });

  it('navbar brand should have href="/" attribute', () => {
    expect(wrapper.find('NavLink.navbar-brand').prop('href')).toEqual('/');
  });

  it('navbar brand should have to="/" attribute', () => {
    expect(wrapper.find('NavLink.navbar-brand').prop('to')).toEqual('/');
  });

  it('should render navbar-trigger inside navbar-header', () => {
    expect(wrapper
      .find('div.navbar-header')
      .find('div.navbar-trigger'))
      .toHaveLength(1);
  });

  it('should change state when the navbar-trigger div is clicked', () => {
    expect(wrapper.state('open')).toEqual(false);
    wrapper.find('div.navbar-trigger').simulate('click');
    expect(wrapper.state('open')).toEqual(true);
  });

  it('should render Search component inside navbar-header', () => {
    expect(wrapper
      .find('div.navbar-header')
      .find('Search'))
      .toHaveLength(1);
  });

  it('should render navbar-logo-img img inside navbar-brand Link', () => {
    expect(wrapper
      .find('Link.navbar-brand')
      .find('img.navbar-logo-img'))
      .toHaveLength(1);
  });

  it('should render navbar-collapse div', () => {
    expect(wrapper.find('div.navbar-collapse')).toHaveLength(1);
  });

  it("should render 2 navbar-nav ul's", () => {
    expect(wrapper.find('ul.navbar-nav')).toHaveLength(2);
  });

  it('should render plugins NavLink', () => {
    expect(wrapper.find('NavLink.navbar-plugins-btn')).toHaveLength(1);
  });

  it('plugins button should have href="/plugins" attribute', () => {
    expect(wrapper.find('NavLink.navbar-plugins-btn').prop('href'))
      .toEqual('/plugins');
  });

  it('plugins button should have to="/plugins" attribute', () => {
    expect(wrapper.find('NavLink.navbar-plugins-btn').prop('to'))
      .toEqual('/plugins');
  });

  it('should render developers NavLink', () => {
    expect(wrapper.find('NavLink.navbar-developers-btn')).toHaveLength(1);
  });

  it('developers button should have href="/developers" attribute', () => {
    expect(wrapper.find('NavLink.navbar-developers-btn').prop('href'))
      .toEqual('/developers');
  });

  it('developers button should have to="/developers" attribute', () => {
    expect(wrapper.find('NavLink.navbar-developers-btn').prop('to'))
      .toEqual('/developers');
  });

  it('should render Sign in button', () => {
    expect(wrapper.find('Button.navbar-signin-btn')).toHaveLength(1);
  });

  it("should change state when navbar-dropdown-btn's are clicked", () => {
    wrapper.find('NavLink.navbar-dropdowns-btn').forEach((NavLink) => {
      expect(wrapper.state('open')).toEqual(false);
      NavLink.simulate('click');
      expect(wrapper.state('open')).toEqual(true);
      wrapper.setState({ open: false });
    });
  });

  it('should change state when enter key is pressed on navbar-trigger', () => {
    expect(wrapper.state('open')).toEqual(false);
    wrapper.find('div.navbar-trigger').simulate('keypress', { key: 'Enter' });
    expect(wrapper.state('open')).toEqual(true);
  });

  it('should change Collapse in attribute according to open state', () => {
    // prop should be false
    let collapse = wrapper.find('Collapse');
    expect(collapse.prop('in')).toEqual(false);

    // set state
    wrapper.setState({ open: true });

    // prop should be true
    collapse = wrapper.find('Collapse');
    expect(collapse.prop('in')).toEqual(true);
  });
});
