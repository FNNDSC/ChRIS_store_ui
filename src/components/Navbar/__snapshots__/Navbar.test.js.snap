// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
